server/users.ts

```diff
@@ -622,6 +622,7 @@ export class User extends Chat.MessageContext {
 	}
 	authAtLeast(minAuth: string, room: BasicChatRoom | null = null) {
 		if (!minAuth || minAuth === ' ') return true;
+		if (minAuth === 'unlocked' && !this.locked && !this.semilocked) return true;
```

Comment:
I think this should be
```suggestion
		if (minAuth === 'unlocked' && (!this.locked || !this.semilocked)) return true;
```
right?